api/v1alpha1/orchestrator_types.go

```diff
@@ -20,149 +20,182 @@ import (
-}
+	// Configuration for Serverless (K-Native) Operator. Optional
+	ServerlessOperator ServerlessOperator `json:"serverlessOperator,omitempty"`
 
-type SonataFlowOperator struct {
-	IsReleaseCandidate bool         `json:"isReleaseCandidate,omitempty"`
-	Enabled            bool         `json:"enabled,omitempty"`
-	Subscription       Subscription `json:"subscription,omitempty"`
-}
+	// Configuration for RHDH (Backstage).
+	// +kubebuilder:validation:Required
+	RHDHConfig RHDHConfig `json:"rhdh"`
 
-type ServerlessOperator struct {
-	Enabled      bool         `json:"enabled,omitempty"`
-	Subscription Subscription `json:"subscription,omitempty"`
-}
+	// Configuration for existing database instance
+	// Used by Data index and Job service
+	// +kubebuilder:validation:Required
+	PostgresDB Postgres `json:"postgres"`
 
-type BackstageSecret struct {
-	BackendSecret string `json:"backendSecret,omitempty"`
-}
+	// Configuration for Orchestrator. Optional
+	ServerlessWorkflow ServerlessWorkflow `json:"orchestrator,omitempty"`
 
-type ClusterTokenUrl struct {
-	ClusterToken string `json:"clusterToken,omitempty"`
-	ClusterUrl   string `json:"clusterUrl,omitempty"`
-}
+	// Configuration for Tekton. Optional
+	Tekton Tekton `json:"tekton,omitempty"`
 
-type GithubBS struct {
-	Token        string `json:"token,omitempty"`
-	ClientID     string `json:"clientId,omitempty"`
-	ClientSecret string `json:"clientSecret,omitempty"`
+	// Configuration for ArgoCD. Optional
+	ArgoCd ArgoCD `json:"argocd,omitempty"`
 }
 
-type ArgoCDBS struct {
-	Enabled   bool   `json:"enabled,omitempty"`
-	Namespace string `json:"namespace,omitempty"`
-	Url       string `json:"url,omitempty"`
-	Username  string `json:"username,omitempty"`
-	Password  string `json:"password,omitempty"`
+type ServerlessLogicOperator struct {
+	// Determines whether to install the ServerlessLogic operator
+	// Defaults to true
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
 }
 
-type NotificationEmailBS struct {
-	Hostname string `json:"hostname,omitempty"`
-	Username string `json:"username,omitempty"`
-	Password string `json:"password,omitempty"`
+type ServerlessOperator struct {
+	// Determines whether to install the Serverless operator
+	// Defaults to true
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
 }
 
-type SecretRefBS struct {
-	Name               string              `json:"name,omitempty"`
-	Backstage          BackstageSecret     `json:"backstage,omitempty"`
-	Github             GithubBS            `json:"github,omitempty"`
-	ClusterTokenUrl    ClusterTokenUrl     `json:"k8s,omitempty"`
-	ArgoCD             ArgoCDBS            `json:"argocd,omitempty"`
-	NotificationsEmail NotificationEmailBS `json:"notificationsEmail,omitempty"`
-}
+type RHDHConfig struct {
+	// Name of RHDH CR, whether existing or to be installed
+	// +kubebuilder:validation:Required
+	RHDHName string `json:"name"`
```

Comment:
we already are in `RHDHConfig` so you do not need to prefix everything by `RHDH`, it's clear already :)

Reply:
I agree, its redundant, updated this to `name` and `namespace`